lib/rules/no-fallthrough.js

```diff
@@ -39,6 +39,19 @@ function isReachable(segment) {
     return segment.reachable;
 }
 
+/**
+ * Checks whether a SwitchCase node is adjacent to the token after it
+ * @param {ASTNode} node - The case node to check
+ * @param {RuleContext} context - Rule context
+ * @returns {boolean} `true` if the case is adjacent to the next token
+ */
+function isAdjacentToNextToken(node, context) {
```

Comment:
I think the term "adjacent" is a bit misleading here. You only care about line numbers, correct? Should this be named something that better reflects what it's doing?

Also, it would better to just pass the SourceCode object in as the second argument rather than the context object.